server/channels/app/user.go

```diff
@@ -255,7 +255,7 @@ func (a *App) createUserOrGuest(c request.CTX, user *model.User, guest bool) (*m
 		case errors.Is(nErr, users.AcceptedDomainError):
 			return nil, model.NewAppError("createUserOrGuest", "api.user.create_user.accepted_domain.app_error", nil, "", http.StatusBadRequest).Wrap(nErr)
 		case errors.As(nErr, &nfErr):
-			return nil, model.NewAppError("createUserOrGuest", "api.user.check_user_password.invalid.app_error", nil, "", http.StatusBadRequest).Wrap(nErr)
+			return nil, model.NewAppError("createUserOrGuest", nfErr.Id(), map[string]interface{}{"Min": *a.Config().PasswordSettings.MinimumLength}, "", http.StatusBadRequest)
```

Comment:
Can't passwords be invalid for reasons other than minimum length?

Reply:
*I'm thinking out loud*

True.

They can be invalid for several reasons:
- doesn't meet min length requirement
-  too long
- doesn't have the right character / numeric character count
- its just invalid/ doesn't match entry in db.

The question is which one of these attributes have not been met.

Reply:
I think I misunderstood this line the first time I read it.

I now see that you are using `nfErr.Id()` so it will use the correct error message.
You are passing in min for the translation since that appears to be required for the all of the misc `model.user.is_valid.pwd` error messages.

It is not required for the `model.user.is_valid.pwd_max_length.app_error` error message. Not sure if it will be required for the other error messages that you have to add (per my other comment)

Reply:
Don't really see this as an issue as long as the tests are passing, I think what is most important is getting the correct error ID.

Reply:
If this is still an pertinent issue this can probably be resolved with an extra case statement just for the `max length password error`.